src/scenarios/scenarios_7.ts

```diff
@@ -0,0 +1,112 @@
+import {ScenarioInterface} from './scenario-interface';
+import Logger from '../services/logger';
+import {SubstrateService} from '../services/substrate-service';
+import cereTypes from './cere-network-type.json';
+const CERE_DECIMAL = 10;
+
+type EntityObject = {
+  validator?: string;
+  validatorStake?: number;
+  nominators?: {who: string; value: number; earned: number}[];
+  eraRewardPoint?: number;
+  validatorPoolEarning?: number;
+  earnedTokenPerStakedToken?: number;
+  totalStake?: number;
+  validatorEarning?: number;
+};
+
+export class Scenarios_7 implements ScenarioInterface {
+  logger = new Logger(Scenarios_7.name);
+
+  public entity: EntityObject[] = [];
+  public totalEraRewardPoints: number;
+  public validatorReward: number;
+  public treasuryReward: number;
+  public tokenPerPoint: number;
+  public eraIndex = +process.env.ERA_INDEX;
+  public blockNumber = +process.env.BLOCK_NUMBER;
+
+  public async run() {
+    this.logger.log(`Starting scenario 7...`);
+
+    this.logger.log(`Initializing the blockchain`);
+    const service = new SubstrateService();
+    await service.initialize(cereTypes);
+
+    await this.fetchValidatorAndNominator(service);
+    await this.eraRewards(service, this.eraIndex);
+    await this.fetchRewards(service);
+
+    this.tokenPerPoint = this.validatorReward / 10 ** CERE_DECIMAL / Number(this.totalEraRewardPoints);
+    await this.calculate();
+
+    this.logger.log(`Report ${JSON.stringify(this.entity)}`);
+  }
+
+  private async fetchValidatorAndNominator(service): Promise<any> {
+    const electedInfo = await service.fetchElectedInfo();
```

Comment:
Validators and nominators should be taken from the previous era.

Reply:
Updated